collaboration-concept.md

```diff
@@ -0,0 +1,137 @@
+# Collaboration Concept
+
+## Microsoft VSCode Live Share Extension
+
+The official Microsoft [Live Share Extension](https://visualstudio.microsoft.com/services/live-share/) implements collaboration support for the following features:
+
+- Editor Syncing (Text + Client markers for each user)
+- Host Syncing (i.e. other users follow the host into their currently selected file + position)
+- Terminal Syncing (if enabled). Both input+output is synced
+- Debug Syncing (if enabled). Both users and host can step through the code as if it were on one machine. Breakpoints are synced as well.
+- Plugin Syncing (i.e. external user has access to all installed plugins/extensions of the host user)
+- Chat Functionality
+- Mechanism for making the API available to other extensions, see [vsls](https://www.npmjs.com/package/vsls)
+
+Similar to the tunnel feature, the connection between host and other users is accomplished using hole-punching.
+I.e. a single central server serves as an intermediary to exchange messages. No messages are being exchanged directly between users. 
+
+## Requirements for Theia Collaboration Feature
+
+### Technical Requirements
+
+The protocol/implementation that this new live share feature is supposed to be using therefore has a few different tasks for different widgets/features
+
+1. Document syncing akin to yjs monaco binding + Awareness information exchange (i.e. broadcast information to all users)
+2. Virtual File System coming from the host user
+3. Frontend message proxying for debug sync (i.e. debug launch or monaco language service requests need to be proxied to the plugin host and the results send back)
+4. Terminal input + output syncing requires a yjs like approach for input and a broadcast for output
+5. Broadcast required for Chat functionality
```

Comment:
Should we also list other potential requirements like:
 - privacy (e.g. allow to configure an own message server without going through one central public instance)
 - encryption
 - authentication